### DISCUSSION TITLE
Desperately need help tuning yolov9 for distant person detection

### DISCUSSION THREAD

**Discussion by Kaelrox** [ultralytics/ultralytics#21476] [Q&A]
Hello Yolo community.

I am working on a project where I am using  both real and synthetic (generated)  image data to train yolov9 to detect people in the water close to the beach.  The view is generally top down from the perspective of a drone. 

The problem is this: No matter what permutation of  augments I use including many epochs, flip, mixup, mosaic etc, my model does not break a box loss level of less than 1.2 or so. It just plateaus after a 150-ish epochs. I'm starting to think there is something fundamentally wrong with my setup. I have attached a sample of the synth data and annotation as well as real image. Any and all help appreciated.

[Man_5_treading_img_0000044.txt](https://github.com/user-attachments/files/21352481/Man_5_treading_img_0000044.txt)
<img width="4056" height="3040" alt="Man_5_treading_img_0000044" src="https://github.com/user-attachments/assets/4fc72274-8bba-4dff-a99b-0678cc12fefc" />
![1158_jpeg_jpg rf c64c54d6d697f7de5cd3e49be14f35ec](https://github.com/user-attachments/assets/720a685e-928a-4045-8aae-e697dbe0ed53)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Kaelrox, thank you for sharing the details of your project and providing data samples! 🚀 This is an automated response to help you get started—an Ultralytics engineer will review your discussion and assist you soon.

For tuning and custom training questions like yours, please include as much information as possible (which you've started to do—thank you!). Helpful details include:
- Example images and annotations (provided ✅)
- Training logs or screenshots of your loss curves and metrics
- Information on your dataset structure and any preprocessing steps
- Confirmation you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)

If you suspect a 🐛 bug, please also provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate.

For more resources and usage guides, check out the [Ultralytics Docs](https://docs.ultralytics.com/), including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples.

Join the Ultralytics community where it suits you best:
- [Discord](https://discord.com/invite/ultralytics) for real-time chat 🎧
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://www.reddit.com/r/Ultralytics/) for community threads

## Upgrade

Make sure you are running the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these up-to-date verified environments (with dependencies like [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for reaching out—someone from the team will dive deeper into your case soon! 😊

**Comment by ExtReMLapin** [comment]
Show us your training args

**Reply by Kaelrox** [reply]
Hey  thank you for reaching out. Right now I am using

results = model.train(data="data.yaml", epochs=300 ,imgsz=640 , batch = 0.80, workers = 32,  warmup_epochs=3, mosaic=1.0, mixup =0.0, scale= 0.0, rect=False, cache=True)

**Reply by ExtReMLapin** [reply]
Image size is too low, increase to something like 1280 or even 2048
Try without mosaic

how can your batch size be lower than 1 ?

**Reply by Kaelrox** [reply]
Hi,  originally did try it with higher image size and it did not perform well. Worse in some cases. Or maybe I was not using large enough image size?

As for  for the  batch size I am using it as per the documentation. When  you provide a floating point value yolo will automatically calculate batch size to use that much gpu memory:

Batch size, with three modes: set as an integer (e.g., batch=16), auto mode for 60% GPU memory utilization (batch=-1), or auto mode with specified utilization fraction (batch=0.70).
Is there something else I could be using?

**Reply by ExtReMLapin** [reply]
Also show results.png image at end of training
